common/dtd/ldmlSupplemental.dtd

```diff
@@ -1154,3 +1154,16 @@ Except as contained in this notice, the name of a copyright holder shall not be
 <!ELEMENT languageGroup ( #PCDATA ) >
 <!ATTLIST languageGroup parent NMTOKEN #REQUIRED >
 	<!--@MATCH:validity/language-->
+
+<!-- # Grammatical Features -->
+<!ELEMENT grammaticalData ( grammaticalFeatures* )>
+
+<!ELEMENT grammaticalFeatures ( grammaticalFeatureValues* ) >
+<!ATTLIST grammaticalFeatures targets ( nominal ) #REQUIRED >
+<!ATTLIST grammaticalFeatures locales NMTOKENS #REQUIRED >
+	<!--@MATCH:validity/language-->
+
+<!ELEMENT grammaticalFeatureValues EMPTY>
+<!ATTLIST grammaticalFeatureValues type ( case | gender | definiteness | state) #REQUIRED >
+<!ATTLIST grammaticalFeatureValues value NMTOKENS #REQUIRED >
```

Comment:
Hmmm. Since the valid values depend on the type, it may be better to split these up, eg

< grammaticalFeatureValues type='case' value=...
=>
< grammaticalCaseValues value=...
that way the values will be specific to case.
(and the same of course for gender, definiteness, state)

Reply:
Thanks! Changed throughout. Should I drop the "grammatical" prefix for inner elements?